generators/server/templates/gradle/profile_dev.gradle.ejs

```diff
@@ -54,7 +58,7 @@ task webpackBuildDev(type: <%= _.upperFirst(clientPackageManager) %>Task) {
 
     outputs.dir("<%= CLIENT_DIST_DIR %>")
 
-    dependsOn <%= clientPackageManager %>Install
+    dependsOn <%= clientPackageManager %>_install
```

Comment:
If i get it right, this means the `install` task will be executed regardless if gradle would consider it up to date? As we already differ between `npm` and `yarn` for the lock file we could also put the correct install task to that block I would say.

Reply:
I do not think so. From what I understand as long as inputs + outputs do not change then the whole `webpackBuildDev` task will be skipped due to incremental compilation taking effect. I believe the declaration of `dependsOn npm_install` should be after the declarations of inputs+ outputs in order for everything to work properly but I can check if you like.

Reply:
Let me give it a try, maybe I am wrong here.

Reply:
@pvliss We should try to use `npmInstall` instead of `npm_install`:

Second run when using `npmInstall`
```
./gradlew webpackBuildDev                                                                                              1.7m  Wed 20 Mar 2019 05:03:48 PM CET
> Task :nodeSetup SKIPPED
> Task :npmSetup SKIPPED
> Task :npmInstall UP-TO-DATE
> Task :webpackBuildDev UP-TO-DATE

BUILD SUCCESSFUL in 2s
2 actionable tasks: 2 up-to-date
```

Second run when using `npm_install` (as basically `npm_install` does nothave proper up-2-date checking it is executed, but does nothing)

```
./gradlew webpackBuildDev                                                                                           34.8s  Wed 20 Mar 2019 05:04:44 PM CET
> Task :nodeSetup SKIPPED
> Task :npmSetup SKIPPED

> Task :npm_install
npm WARN acorn-dynamic-import@4.0.0 requires a peer of acorn@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.3.1 requires a peer of jquery@1.9.1 - 3 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.3.1 requires a peer of popper.js@^1.14.7 but none is installed. You must install peer dependencies yourself.
npm WARN jest-preset-angular@6.0.2 requires a peer of jest@^23.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN ts-jest@23.1.4 requires a peer of babel-core@^6.0.0 || ^7.0.0-0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.7 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.7: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

up to date in 28.619s

> Task :webpackBuildDev UP-TO-DATE

BUILD SUCCESSFUL in 37s
2 actionable tasks: 1 executed, 1 up-to-date
```

Reply:
@pvliss We should try to use `npmInstall` instead of `npm_install`:

Second run when using `npmInstall`
```
./gradlew webpackBuildDev                                                                                              1.7m  Wed 20 Mar 2019 05:03:48 PM CET
> Task :nodeSetup SKIPPED
> Task :npmSetup SKIPPED
> Task :npmInstall UP-TO-DATE
> Task :webpackBuildDev UP-TO-DATE

BUILD SUCCESSFUL in 2s
2 actionable tasks: 2 up-to-date
```

Second run when using `npm_install` (as basically `npm_install` does nothave proper up-2-date checking it is executed, but does nothing)

```
./gradlew webpackBuildDev                                                                                           34.8s  Wed 20 Mar 2019 05:04:44 PM CET
> Task :nodeSetup SKIPPED
> Task :npmSetup SKIPPED

> Task :npm_install
npm WARN acorn-dynamic-import@4.0.0 requires a peer of acorn@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.3.1 requires a peer of jquery@1.9.1 - 3 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.3.1 requires a peer of popper.js@^1.14.7 but none is installed. You must install peer dependencies yourself.
npm WARN jest-preset-angular@6.0.2 requires a peer of jest@^23.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN ts-jest@23.1.4 requires a peer of babel-core@^6.0.0 || ^7.0.0-0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.7 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.7: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

up to date in 28.619s

> Task :webpackBuildDev UP-TO-DATE

BUILD SUCCESSFUL in 37s
2 actionable tasks: 1 executed, 1 up-to-date
```

Reply:
Good find. Let me update the PR then.

Reply:
Sorry for double posting, but the connection currently is really bad.

Reply:
No problem